src/MOI_wrapper.jl

```diff
@@ -37,32 +37,46 @@ const SUPPORTED_CONSTRAINTS = [
 
 mutable struct Optimizer <: LQOI.LinQuadOptimizer
     LQOI.@LinQuadOptimizerBase(Model)
-    env::Env
-    params::Dict{String,Any}
-    Optimizer(::Nothing) = new()
+    env::Union{Nothing, Env}
+    params::Dict{String, Any}
+
+    """
+        Optimizer(env = nothing; kwargs...)
+
+    Create a new Optimizer object.
+
+    You can share Gurobi `Env`s between models by passing an instance of `Env`
+    as the first argument. By default, a new environment is created for every
+    model.
+
+    Note that we set the parameter `InfUnbdInfo` to `1` rather than the default
+    of `0` so that we can query infeasibility certificates. Users are, however,
+    free to overide this as follows `Gurobi.Optimizer(InfUndbInfo=0)`.
+    """
+    function Optimizer(env::Union{Nothing, Env} = nothing; kwargs...)
+        model = new()
+        model.env = env
+        model.params = Dict{String, Any}()
+        MOI.empty!(model)
+        for (name, value) in kwargs
+            model.params[string(name)] = value
+            setparam!(model.inner, string(name), value)
```

Comment:
What happens if the name is not a valid error ? Does Gurobi throws an helpful error ?
We should maybe add a comment about that

Reply:
`Gurobi.Optimizer(bad_kwarg = 1)` gives `Error: Unrecognized parameter name: bad_kwarg.` So that seems fine for now.